File: Consecutive 1's not allowed - GFG/consecutive-1s-not-allowed.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>

using namespace std;
#define ll long long

// } Driver Code Ends
//User function template for C++
class Solution{
public:
	// #define ll long long
	ll countStrings(int n) {
	    ll zeroEnd=1;
	    ll oneEnd=1;
	    ll sum = zeroEnd+oneEnd;
	    if(n==1){
	        return sum;
	    }
	    ll i=2;
	    while(i<=n){
	        oneEnd=zeroEnd%1000000007;
	        zeroEnd=sum%1000000007;
	        sum=(oneEnd+zeroEnd)%1000000007;
	        i++;
	    }
	    return sum;
	}
};

//{ Driver Code Starts.

int main() {
    int t;
    cin >> t;
    while (t--) {
        int n;
        cin >> n;
        Solution ob;
        auto ans = ob.countStrings(n);
        cout << ans << "\n";
    }
    return 0;
}
// } Driver Code Ends

File: Consecutive 1's not allowed - GFG/consecutive-1s-not-allowed.java
//{ Driver Code Starts
// Initial Template for Java

import java.io.*;
import java.util.*;

class GFG {
    // Driver code
    public static void main(String[] args) throws Exception {
        BufferedReader br =
            new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(br.readLine().trim());
        while (t-- > 0) {
            int n = Integer.parseInt(br.readLine().trim());

            long ans = new Solution().countStrings(n);

            System.out.println(ans);
        }
    }
}
// } Driver Code Ends


// User function Template for Java

class Solution {
    long countStrings(int n) {
        long zeroEnd = 1;
        long oneEnd = 1;
        long sum = zeroEnd+oneEnd;
        if(n==1){
            return sum;
        }
        long i=2;
        while(i<=n){
            oneEnd = zeroEnd%1000000007;
            zeroEnd = sum%1000000007;
            sum = (zeroEnd + oneEnd)%1000000007;
            i++;
        }
        return sum%1000000007;
    }
}